### ISSUE TITLE
Slashed zero variant?

### ISSUE THREAD

**Issue by NetOperatorWibby** [JetBrains/JetBrainsMono#8]
This typeface looks phenomenal but I much prefer to have a slash in my zero versus a dot inside it.

**Comment by daniellizik** [comment]
On osx vscode integrated terminal `0` is nearly indistinguishable from `8`

![Screen Shot 2020-01-16 at 16 46 46](https://user-images.githubusercontent.com/10506366/72503737-d132f080-387f-11ea-8618-901ebd005bc4.png)

**Comment by NetOperatorWibby** [comment]
@daniellizik There is no slash currently, it's a dot. Also, I think increasing your text size would help discern the difference.

The reason why the zero has anything in it to begin with is to better distinguish from "O". Monospaced "O" looks awfully similar to "0" at a glance.

**Comment by philippnurullin** [comment]
There is a slashed zero in JetBrains Mono. But to use it, your IDE needs to support stylistic sets (opentype feature).

**Comment by NetOperatorWibby** [comment]
@philippnurullin Thanks for the tip! Sublime Text apparently doesn't have support for specifying a slashed zero in its settings. Still.

**Comment by NetOperatorWibby** [comment]
If your IDE doesn't support the ligatures necessary for slashed zeroes, [this patch](https://github.com/JetBrains/JetBrainsMono/files/4091880/OpenType-TT.zip) should work. I used FontLab to replace the default zero with the slashed variant. In FontBook (macOS) you won't see it in the preview but it'll appear in your IDE.

**Comment by qgates** [comment]
@philippnurullin I'd like this option too, but I can't see how in sublime. Be nice to have a variant of the font pack with slashed zeros for users of older software / terminals etc..

Vscode can use stylistic sets however, just change `editor.fontLigatures` setting (JSON settings):

```
"editor.fontLigatures": "'zero'",
```

**Comment by sewkokot** [comment]
Yes, it would be good to be able to download the font pack with slashed zero as default. Emacs don't support stylistic sets either. The dotted zero is somewhat unpredictable, sometimes the dot is centered and in other font sizes is not. For me JBM font wins ahead of Consolas (my so far winner), Hack, Fira Code, Fantasque Sans Mono and Input Mono. Thank you :)

**Comment by k0pernikus** [comment]
@philippnurullin Do JetBrains IDE support "support stylistic sets (opentype feature)", and if so, how do I enable this?

**Comment by qgates** [comment]
@sewkokot +1 on all your points. FWIW I created a separate issue #104 about dotted 0 being off-center at different font sizes.

**Comment by philippnurullin** [comment]
@k0pernikus Unfortunately not yet. You can follow this issue on YouTrack https://youtrack.jetbrains.com/issue/IDEA-221886

**Comment by philippnurullin** [comment]
@qgates Wow, sorry. Totally missed your message. 

There is not much beside the slashed zero feature now.

Making a lot of different font family's is not a good solution. For example, is we will make the version of JetBrains Mono without ligatures in 4 weights with italics, 8 total + version with ligatures, 8 files. Then each of this version will have the dotted zero & slashed. Thats leaves us with  32 font files. At some point (soon) we will be expanding the weights from 4 to 7, its 56 files. Yes, it can be automated, but complexity of system will only grow & create an opportunity for an error. The only way i see it can be done good, is adopting the approach of Input Mono with [the visual font family settings.](https://input.fontbureau.com/preview/?size=14&language=python&theme=solarized-dark&family=InputSans&width=300&weight=300&line-height=1.2&a=0&g=0&i=0&l=0&zero=0&asterisk=0&braces=0&preset=default&customize=please) This will be a good "next" step, right now we are more concentrated on present.

**Comment by qgates** [comment]
@philippnurullin Hey, no apology necessary :slightly_smiling_face:

I agree that having multiple fonts for different tastes could easily get out of hand and I also like how Input Mono has solved that problem. This would be great for the future, and might allow for a bunch of variations including the `$` discussed elsewhere, zero and also `{}` which I think could look slightly nicer in JBM (don't worry, not going to create an issue about this!).

Oh and while on the nitpick there's something about the `5` that looks slightly wrong as well. Think the bottom loop is slightly overcome by the square top. Fira has that one perfect :wink:

**Comment by djoerg** [comment]
It seems to me, that the need to distinguish the 'O' from zero is a relict from the past where all IDEs and Text Editors had only a single color.

Consider this:
 - almost all IDEs/TextEditors these days colorize numers and characters differently
 - how often do you have an 'O' next to a zero and how often a '8' next to a zero?

For me, that are two very good reasons to not decorate the zero with anything that makes it harder to distinguish from an '8'!

@NetOperatorWibby May i kindly ask you to invest the time to make a version without any decoration on the zero? (Maybe just copy the 'O'). I would highly appreciate it, and i think many others too.

**Comment by kovica** [comment]
@djoerg No, we won't.

Don't you guys mess with my 0s and Os. :)

**Comment by KillyMXI** [comment]
Interesting. How in the world I'm supposed to learn that this font has a stylistic set (OpenType alternative) for zero looks?

It has to be documented somewhere.

But, while I'm happy the issue with the slashed zero is somewhat addressed, it seems like I can't make use of it in VScode. Using the `zero` feature also means I have to enable ligatures. But I can't stand them before the issue #11 is resolved.

From `https://fontdrop.info/` I can see that there are two features - `zero` and `calt`. But any combination for the `"editor.fontLigatures"` value except `false` leaves all ligatures enabled, while I need all shape-changing garbage disabled.

**Comment by alexeyten** [comment]
@KillyMXI 
```json
"editor.fontLigatures": "'calt' 0, 'zero' 1"
```

![image](https://user-images.githubusercontent.com/44377/75809116-49eefd00-5d99-11ea-95b9-a45c22f287fc.png)

**Comment by KillyMXI** [comment]
@alexeyten thank you! It did the trick.

Now my only complain within this issue is about documentation - better have all available features listed in the readme / at the website.

**Comment by philippnurullin** [comment]
@KillyMXI Yes, we will add the list of all the features at some point. 
Right now there is only this one, so it fells kind stupid to add the whole section with OpenType features list. But i understand why it can be important.

**Comment by BornForFever** [comment]
@NetOperatorWibby If you prefer shashed zero, set 'zero' to your font. For example, if you're using VS Code, add `"editor.fontLigatures": "'zero'"` to the `settings.json` file.

**Comment by NetOperatorWibby** [comment]
> @philippnurullin Thanks for the tip! **Sublime Text** apparently doesn't have support for specifying a slashed zero in its settings. Still.

**Comment by chiniara** [comment]
> 
> 
> @NetOperatorWibby If you prefer shashed zero, set 'zero' stylistic feature to your font. For example, if you're using VS Code, add `"editor.fontLigatures": "'zero'"` to the `settings.json` file.

How would i do that in visual studio 2019?

The problem is that most of the tools doesnt support stylistic sets, dotted zeroes are hard to distinguish at small sizes.
The font already has a no ligature version, an slashed zero variant would be nice to have too.
The dotted zero is the only thing that is making me not use jetbrains mono

**Comment by cata0309** [comment]
@philippnurullin can you provide a version with slashed zero ? I am using Intellij IDEA based IDE's and I can't get around this issue.

**Comment by philippnurullin** [comment]
@cata0309 Hi. I can't because of the the problems it will bring. More about it here → https://github.com/JetBrains/JetBrainsMono/issues/8#issuecomment-584134324
In IDEA behalf you can follow this issue https://youtrack.jetbrains.com/issue/IDEA-221886

**Comment by cata0309** [comment]
I editted the font in FontForge by switching the zeros, I had no problem using it thought.

**Comment by philippnurullin** [comment]
@cata0309 Glad to hear you fond a solution.

**Comment by k0pernikus** [comment]
@cata0309 Can you please provide your patched font file?

**Comment by cata0309** [comment]
Of course, is it ok if I link it here @philippnurullin ? I am thinking about making an archive.

**Comment by cata0309** [comment]
Ok, so I swapped the dotted zero with slashed zero(and also added slashed zero to the NL versions) 
[JetBrains-Mono_Zero.zip](https://github.com/JetBrains/JetBrainsMono/files/4458636/JetBrains-Mono_Zero.zip)
Note 1: I do not renamed the actual name font !!! In order for it to work you have to remove the JetBrains Mono family before installing the patched one!
Note 2: Let me know if it's ok to you that I shared the modified fonts here

**Comment by chiniara** [comment]
@cata0309 loved it ! im already using this version

**Comment by philippnurullin** [comment]
@cata0309 I'm 100% ok with this. Tat's why the project is open source.

**Comment by cata0309** [comment]
I am glad I made someone happy !

**Comment by lostcoda** [comment]
@cata0309 Thanks for the patched version, but the ligatures have seemed to stop working.

**Comment by lostcoda** [comment]
@cata0309 Thanks for the patched version, but the ligatures seem to have stopped working.

**Comment by NetOperatorWibby** [comment]
@lostcoda I made [my own patch](https://github.com/JetBrains/JetBrainsMono/issues/8#issuecomment-576746700) a few days after I made this issue and ligatures work.

**Comment by lostcoda** [comment]
@NetOperatorWibby In Intellij, ligatures work just fine from your patch but slashed zero doesn't. It is still dotted.

**Comment by NetOperatorWibby** [comment]
@lostcoda That's strange, it works in Sublime Text. Did you completely remove the previous font before installing my patched one?

**Comment by lostcoda** [comment]
@NetOperatorWibby Yes, I did. Everything works fine as it is in vscode, but not in intellij.

**Comment by NetOperatorWibby** [comment]
@lostcoda Oh damn, idk what to do about that.

**Comment by trygveaa** [comment]
IntelliJ bundles JetBrains Mono. I guess you have to remove it from the IntelliJ installation in addition to removing it globally.

**Comment by philippnurullin** [comment]
@lostcoda You can upgrade to 2020.1 or manually remove the bundled fonts. This guid will give you an idea where to find them https://github.com/JetBrains/JetBrainsMono/issues/142

**Comment by lostcoda** [comment]
@trygveaa @philippnurullin Everything's working fine after removing the built-in bundle. Thanks.

**Comment by sharpjs** [comment]
I've made available [JetBrains Mono Slashed](https://github.com/sharpjs/JetBrainsMonoSlashed), which is JetBrains Mono with slashed zero as the default.  I am using the font myself, so I will try to keep it relatively up-to-date with the official font.

**Comment by Ypnose** [comment]
Thanks for your work @sharpjs !
It is much better, especially for people subject to astigmatism.

**Comment by grb0** [comment]
@sharpjs 
Actually your slashed version renders some characters slightly different than the original version: e, f, J, K, % to name a few.
If you could investigate what's going on and have slashed version rendering identically to the original version.

Thank you and have a good one!

**Comment by elshize** [comment]
For posterity and search indexer: in kitty terminal, here's what you do:
```
font_features JetBrainsMono-Light +zero
```
[More info in docs](https://sw.kovidgoyal.net/kitty/conf.html#opt-kitty.font_features)

**Comment by szeiger** [comment]
To add a data point: This is in the latest IntelliJ on Windows 10, 15 points on a screen with 125% dpi scaling. 0 and 8 are almost indistinguishable (which IMHO is far worse than 0 and O):
<img width="101" alt="Screenshot 2021-11-11 181719" src="https://user-images.githubusercontent.com/54262/141340875-560a2d91-a85e-41c6-82a9-a00146ce8e69.png">

**Comment by k0pernikus** [comment]
> [@philippnurullin](https://github.com/philippnurullin) Do JetBrains IDE support "support stylistic sets (opentype feature)", and if so, how do I enable this?

It arrvied! It is here! (EAP version at least, e.g. pycharm@2026.1-261.17801.91)

Settings > Font > Typographic Settings, Character Variants

<img width="1784" height="923" alt="Image" src="https://github.com/user-attachments/assets/a752e54f-1daa-4448-b5b1-3b732456d16e" />

It is so beautiful! (Screenshot showcasing Cascadia Code as I had other fonts patched on my machine to deal with Jetbrains lack of stylistic support. The wait is finally over. Finally, https://youtrack.jetbrains.com/issue/IJPL-26382/Support-stylistic-sets-for-fonts was implemented.

---

I can showcase JetBrains Mono just the same, patched or not:

<img width="1357" height="1154" alt="Image" src="https://github.com/user-attachments/assets/7dbea08c-66d4-4291-b796-f2cecc7563de" />
